src/item.cpp

```diff
@@ -825,7 +825,41 @@ std::string Item::getDescription(const ItemType& it, int32_t lookDistance,
 				tmpSubType = item->getSubType();
 			}
 
-			s << ". " << (it.stackable && tmpSubType > 1 ? "They" : "It") << " can only be used with";
+			s << ". " << (it.stackable && tmpSubType > 1 ? "They" : "It") << " can only be used by ";
+
+			VocSpellMap vocMap = g_spells->getRuneSpell(it.id)->getVocMap();
+			if (vocMap.empty()) {
+				s << "players";
+			} else {
+				std::vector<Vocation*> showVocMap;
+
+				// vocations listed are mostly unpromoted ones and the promoted version, with the latter hidden from
+				// description, so probably always `total / 2` is the amount of vocations to be shown.
+				showVocMap.reserve(vocMap.size() / 2);
+
+				for (const auto& voc: vocMap) {
+					if (voc.second) {
+						showVocMap.push_back(g_vocations.getVocation(voc.first));
+					}
+				}
+
+				auto vocIt = showVocMap.begin(), vocLast = (showVocMap.end() - 1);
+
+				while (vocIt != vocLast) {
+					auto vocName = asLowerCaseString((*vocIt)->getVocName());
```

Comment:
Append the result from asLowerCaseString directly into the stringstream instead of holding it in a variable first, as it may result in copying it more than we need to (make sure the iterator is incremented after appending it and not before).